packages/statemanager/test/flat/snapshot.spec.ts

```diff
@@ -0,0 +1,258 @@
+import { Trie } from '@ethereumjs/trie'
+import { Account, Address, KECCAK256_RLP, bytesToHex, hexToBytes } from '@ethereumjs/util'
+import { keccak256 } from 'ethereum-cryptography/keccak'
+import { assert, describe, it } from 'vitest'
+
+import { FlatStateManager, Snapshot } from '../../src/index.js'
+;(BigInt.prototype as any).toJSON = function () {
+  return this.toString()
+}
+
+async function merkleizeViaTrie(leaves: [Address, Uint8Array][]) {
+  const trie = new Trie({ useKeyHashing: true })
+  for (const leaf of leaves) {
+    await trie.put(leaf[0].bytes, leaf[1])
+  }
+  return trie.root()
+}
+
+describe('snapshot simple get/put', () => {
+  it('should return null for non-existent key', async () => {
+    const snapshot = new Snapshot()
+    const addr = new Address(hexToBytes('0x' + '3'.repeat(40)))
+    const res = await snapshot.getAccount(addr)
+    assert.equal(res, null)
+  })
+
+  it('should put/get account', async () => {
+    const snapshot = new Snapshot()
+    const addr = new Address(hexToBytes('0x' + '3'.repeat(40)))
+    const val = new Account()
+    await snapshot.putAccount(addr, val)
+    const res = await snapshot.getAccount(addr)
+    assert.equal(JSON.stringify(res), JSON.stringify(val.serialize()))
```

Comment:
```suggestion
    assert.deepEqual(res, val.serialize())
```

Reply:
This same format can be applied wherever you are comparing bytes but stringifying first.

Reply:
Fixed this.